dynamorm/model.py

```diff
@@ -417,14 +417,34 @@ def to_dict(self, native=False):
                 pass
         return self.Schema.dynamorm_validate(obj, native=native)
 
-    def save(self, **kwargs):
+    def save(self, partial=False, **kwargs):
         """Save this instance to the table
 
+        :param bool partial: When False the whole document will be ``.put`` to the table.  When True only values that
+                             have changed since the document was loaded will sent to the table via an ``.update``.
+        :param \*\*kwargs: When partial is False these are passed through to the put method on the table.  When partial
+                           is True these become the kwargs for update_item.  See ``.put`` & ``.update`` for more
+                           details.
+
         The attributes on the item go through validation, so this may raise :class:`ValidationError`.
         """
-        # XXX TODO: do partial updates if we know the item already exists, right now we just blindly put the whole
-        # XXX TODO: item on every save
-        return self.put(self.to_dict(), **kwargs)
+        if not partial:
+            return self.put(self.to_dict(), **kwargs)
+
+        # Collect the fields to updated based on what's changed
+        # XXX: Deeply nested data will still put the whole top-most object that has changed
+        # TODO: Support the __ syntax to do deeply nested updates
+        updates = dict(
+            (k, v)
```

Comment:
I believe this should be `(k, getattr(self, k))`?

cc @borgstrom @robby-bryant

Reply:
I can't figure out why the test isn't failing, but my understanding is that updated data should be sitting on the object, but not yet copied to `_validated_data`.

Also, if that is correct, `.update` should set `_validated_data` such that subsequent calls with no change do not attempt to update anything

Reply:
You're correct.  The test is broken.  New PR incoming.